lib/webauthn/authenticator_attestation_response.rb

```diff
@@ -50,6 +50,17 @@ def attestation
       @attestation ||= CBOR.decode(attestation_object)
     end
 
+    def aaguid
+      raw_aaguid = authenticator_data.attested_credential_data.raw_aaguid
+      unless raw_aaguid == WebAuthn::AuthenticatorData::AttestedCredentialData::ZEROED_AAGUID
+        authenticator_data.attested_credential_data.aaguid
+      end
+    end
+
+    def attestation_certificate_key
+      raw_subject_key_identifier(attestation_statement.attestation_certificate)&.unpack("H*")&.[](0)
+    end
```

Comment:
To answer your earlier question @grzuy - my reason for exposing these here is twofold :)
- Access during registration ceremony for metadata lookup
- To be stored, so the metadata can be looked up later (during authentication or periodically in a background job)

In both cases the RP can evaluate whether the authenticator is acceptable at any given moment.

Reply:
Is this only valid for `fido-u2f` fmt?

Do you have sources talking about this in particular. Did a quick search and could find anything.

Reply:
Also, should this be part of the Metadata support PR?
Or is this already providing value because it's giving easier access to users who what to perform the metadata lookup themselves?

Reply:
Yes, only for U2F: sources are [FIDO Alliance MDS specs](https://fidoalliance.org/specs/fido-v2.0-rd-20180702/fido-metadata-service-v2.0-rd-20180702.html#idl-def-MetadataTOCPayloadEntry) and the [part of RFC5280](https://tools.ietf.org/html/rfc5280#section-4.2.1.2) it references.

Since the metadata PR is pretty large and complex, I've been trying to split off PRs that add some value in itself and land those first. But it can be hard to see the total picture that way, and to be honest with the back and forth here and the metadata PR I've also started thinking of pushing this and the metadata handling up into the `WebAuthn::AttestationStatement::Base` subclasses. 

I noticed that the MDS database is pretty empty at the moment: Android SafetyNet, Android keystore, [Windows Hello](https://docs.microsoft.com/en-us/microsoft-edge/dev-guide/windows-integration/web-authentication#implementation-notes) and [Yubikeys](https://support.yubico.com/support/solutions/articles/15000014219-yubikey-5-series-technical-manual#AAGUID_Valuesaledv4) entries are all not present despite being certified. Or in other words: Feitian is the only hardware manufacturer I recognize.

Given that we already verify the validity of Android SafetyNet and Keystore with Google's root certificates, we don't need to consult the FIDO MDS to do the same thing again. The standard [mentions](https://www.w3.org/TR/webauthn/#registering-a-new-credential) MDS is just one way of doing things:

> 15. If validation is successful, obtain a list of acceptable trust anchors (attestation root certificates or ECDAA-Issuer public keys) for that attestation type and attestation statement format _fmt_, from a trusted source or from policy. For example, the FIDO Metadata Service provides one way to obtain such information, using the `aaguid` in the `attestedCredentialData` in _authData_.

 That would keep fido-u2f, packed and tpm to use the MDS, either with `aaguid` or `attestationCertificateIdentifier` (subjectKeyIdentifier).

Happy to chat on gitter or hop on a Skype call if that makes it easier to hash out the details :)

Reply:
Gotcha.

I'm good with either the current implementation or pushing it to the attestation statement subclasses.
Your call.

Thanks for the detailed explanation.